spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/servlet/WebMvcAutoConfiguration.java

```diff
@@ -332,14 +332,20 @@ public void addResourceHandlers(ResourceHandlerRegistry registry) {
 				return;
 			}
 			Duration cachePeriod = this.resourceProperties.getCache().getPeriod();
-			CacheControl cacheControl = this.resourceProperties.getCache()
+			ResourceProperties.Cache.Cachecontrol cacheControl = this.resourceProperties
+					.getCache().getCachecontrol();
+			if (cachePeriod != null && cacheControl.getMaxAge() == null) {
```

Comment:
`cachePeriod` is not only about max-age; if `cachePeriod=-1` nothing is done, `0` means a `no-store` configuration, a positive number results in a max-age configuration. But this should delegate to Spring Framework and not hardcode those assumptions here.
A short term solution would be to check whether `CacheControl.getHeaderValue()` is not `null` and only apply it in this case. A better solution IMO would be to create a Spring Framework issue and getting this solved at the Framework level (see first comment in #16488).

Reply:
Sounds really reasonable 👍 
I think we just need to keep in mind that Webflux has basically the same implementation and when resolved it should also getting fixed there. See: https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/reactive/WebFluxAutoConfiguration.java#L169-L177